### PR TITLE
Gui improvements 08 11 23

### PR DESCRIPTION
**Pull request by maryvictol**
The PR provides updates:
- updates DetachedConfigurationsTest, NfsDataStorageTest
- fix SLURMMasterSetup message
- associate different description estimated price to common primitive ESTIMATED_PRICE
- add `Price cannot be estimated` case handling
- add restoring filters values after refresh  in the getInfoRow method for SystemLoggingTest
- update clearAndFillPipelineFile method

### CODE REVIEW

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchLimitMountsTest.java [hunk 1]
```diff
@@ -192,7 +192,8 @@ public void rerunPipelineWithoutLimitMounts() {
                 .ensure(LIMIT_MOUNTS, text("All available non-sensitive storages"))
                 .launch(this)
                 .showLog(getLastRunId())
-                .ensureNotVisible(PARAMETERS)
+                .expandTab(PARAMETERS)
+                .ensureParameterIsNotPresent("CP_CAP_LIMIT_MOUNTS")
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchLimitMountsTest.java]: Could we extract "CP_CAP_LIMIT_MOUNTS" to constant and replace this

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/RunPipelineTest.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import static com.codeborne.selenide.Selectors.byClassName;
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/RunPipelineTest.java]: Clean up the unused import statements, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/SamplesMetadataTest.java [hunk 3]
```diff
@@ -404,16 +404,19 @@ public void preparePipeline() {
                 .saveAndCommitWithMessage("test: sample metadata")
                 .click(DELETE)
                 .click(button("OK"))
+                .sleep(10, SECONDS)
                 .uploadFile(getFile(launchScript))
                 .ensure(byText(launchScript), visible);
-        sleep(20, SECONDS);
+//        sleep(20, SECONDS);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/SamplesMetadataTest.java]: Could this be cleaned?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/SystemLoggingTest.java [hunk 4]
```diff
@@ -142,15 +146,15 @@ public void userAuthenticationUnsuccessfulAttempt() {
                     .switchToSystemLogs();
             if (impersonateMode()) {
                 systemLogsAO
-                        .filterByUser(admin.login)
+                        .filterByField("User", admin.login)
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/SystemLoggingTest.java]: Let's extract "User" to constant and replace its everywhere

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/LogAO.java [hunk 5]
```diff
@@ -461,6 +461,7 @@ public static By parameterWithName(final String name, final String value) {
                 ".//td[contains(., '%s')]]", name, value));
     }
 
+
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/LogAO.java:468]: Could be cleaned up?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 6]
```diff
@@ -86,28 +86,31 @@ public PipelineCodeTabAO clearAndFillPipelineFile(String fileName, String newTex
...
-        $$(".pipeline-code-form__button").findBy(text("Edit")).shouldBe(exist).click();
 
-        sleep(500, MILLISECONDS);
-        Actions action = actions().moveToElement($(byClassName("CodeMirror-line"))).click();
-        for (int i = 0; i < 1000; i++) {
-            action.sendKeys("\b").sendKeys(Keys.DELETE);
-        }
-        action.perform();
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Edit")).waitUntil(exist, DEFAULT_TIMEOUT).click();
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Could this to be extracted to EDIT primitive?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 7]
```diff
@@ -86,28 +86,31 @@ public PipelineCodeTabAO clearAndFillPipelineFile(String fileName, String newTex
...
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Save")).waitUntil(exist, DEFAULT_TIMEOUT);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Could this to be extracted to SAVE primitive?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 8]
```diff
@@ -86,28 +86,31 @@ public PipelineCodeTabAO clearAndFillPipelineFile(String fileName, String newTex
...
+        Utils.selectAllAndClearTextField(editor);
+        Utils.clickAndSendKeysWithSlashes(editor, newText);
 
-        $$(".pipeline-code-form__button").findBy(text("Save")).click();
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Save")).click();
         $("#message").setValue("test commit message");
         $$("button").findBy(text("Commit")).click();
-        $("ant-modal-content").waitUntil(not(exist), DEFAULT_TIMEOUT);
+        $(className("ant-modal-content")).waitUntil(not(exist), DEFAULT_TIMEOUT);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Could this to be extracted to CONTENT primitive?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 9]
```diff
@@ -86,28 +86,31 @@ public PipelineCodeTabAO clearAndFillPipelineFile(String fileName, String newTex
...
-        Utils.clickAndSendKeysWithSlashes($(byClassName("CodeMirror-line")), newText);
+        SelenideElement editor = $(byClassName("CodeMirror-code"));
+        sleep(500, MILLISECONDS);
+        editor.shouldBe();
+        Utils.selectAllAndClearTextField(editor);
+        Utils.clickAndSendKeysWithSlashes(editor, newText);
 
-        $$(".pipeline-code-form__button").findBy(text("Save")).click();
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Save")).click();
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: The same as above

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 10]
```diff
@@ -86,28 +86,31 @@ public PipelineCodeTabAO clearAndFillPipelineFile(String fileName, String newTex
...
-        action.perform();
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Edit")).waitUntil(exist, DEFAULT_TIMEOUT).click();
+        $(className("ant-modal-content")).$$(".pipeline-code-form__button")
+                .findBy(text("Save")).waitUntil(exist, DEFAULT_TIMEOUT);
 
-        Utils.clickAndSendKeysWithSlashes($(byClassName("CodeMirror-line")), newText);
+        SelenideElement editor = $(byClassName("CodeMirror-code"));
+        sleep(500, MILLISECONDS);
+        editor.shouldBe();
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Could this to be extracted to EDITOR primitive?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java [hunk 11]
```diff
@@ -196,6 +199,18 @@ public PipelineCodeTabAO shouldContainFile(final String fileName) {
         return this;
     }
 
+    public PipelineCodeTabAO waitUntilSaveEnding() {
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Can be private

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java [hunk 12]
```diff
@@ -15,7 +15,9 @@
  */
 package com.epam.pipeline.autotests.ao;
 
+import static com.codeborne.selenide.Condition.exist;
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java]: Clean up the unused import statements, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java [hunk 13]
```diff
@@ -15,7 +15,9 @@
...
 import com.codeborne.selenide.SelenideElement;
+import static com.epam.pipeline.autotests.utils.C.DEFAULT_TIMEOUT;
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java]: Clean up the unused import statements, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineSelection.java [hunk 14]
```diff
@@ -50,6 +53,7 @@ public PipelineSelection(final Configuration parentAO) {
 
     public PipelineSelection selectPipeline(final String pipeline) {
         this.pipeline = pipeline;
+        context().$(className("ant-table-content")).waitUntil(exist, DEFAULT_TIMEOUT);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineSelection.java]: Could this to be extracted to TABLE primitive?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SystemManagementAO.java [hunk 15]
```diff
@@ -105,25 +117,42 @@ public SelenideElement getInfoRow(final String message, final String user, final
...
+            return $(filterBy(filterName))
+                    .$$(By.className("ant-select-selection__choice")).texts();
         }
 
-        public SystemLogsAO filterByMessage(final String message) {
-            setValue(inputOf(filterBy("Message")), message);
-            pressEnter();
+        public String getMessageFilter() {
+            return $(filterBy("Message")).$(byXpath("./input"))
+                    .getAttribute("value");
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SystemManagementAO.java:127]: Could this be better to extract filters to primitives?

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchLimitMountsTest.java (hunk 1 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchLimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/RunPipelineTest.java (hunk 2 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/RunPipelineTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/SystemLoggingTest.java (hunk 4 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/SystemLoggingTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/LogAO.java (hunk 5 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/LogAO.java:468]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/SamplesMetadataTest.java (hunk 3 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/SamplesMetadataTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 6 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 7 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 9 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java (hunk 13 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java (hunk 12 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineGraphTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 11 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 10 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineSelection.java (hunk 14 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineSelection.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java (hunk 8 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/PipelineCodeTabAO.java]: Done

**Review by kamyshova — APPROVED** [review]

**Pull request merged**